src/vmm/src/resources.rs

```diff
@@ -286,7 +317,7 @@ impl VmResources {
         self.vm_config.mem_size_mib = mem_size_mib;
 
         // Update the CPU template
-        if let Some(cpu_template) = machine_config.cpu_template {
+        if let Some(cpu_template) = machine_config.cpu_template.clone() {
             self.vm_config.cpu_template = cpu_template;
         }
```

Comment:
I find doing operations in conditions to be detrimental to readability, I would instead suggest something like:
```suggestion
        if let Some(cpu_template) = machine_config.cpu_template {
            self.vm_config.cpu_template = cpu_template.clone();
        }
```
You can also use `if let Some(cpu_template) = &machine_config.cpu_template` or `if let Some(ref cpu_template) = machine_config.cpu_template` if the aforementioned doesn't work.